eng/Directory.Build.Common.props

```diff
@@ -74,7 +74,7 @@
     -->
     <NoWarn>
       $(NoWarn);
-      AZURE0001; <!-- BicepModelReaderWriterOptions -->
+      AzProvision001; <!-- BicepModelReaderWriterOptions -->
```

Comment:
Nit - maybe update this comment to reference "all of Azure.Provisioning's experimental features" or something broader than than type?

Reply:
Would we reuse this same diagnostic ID for all APIs?